doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/gpu.rst

```diff
@@ -0,0 +1,200 @@
+        containers:
+         - name: ray-node
+           image: rayproject/ray-ml:2.0.0-gpu
+           ...
+           resources:
+            cpu: 3
+            memory: 50Gi
+            nvidia.com/gpu: 1 # Optional, included just for documentation.
+           limits:
+            cpu: 3
+            memory: 50Gi
+            nvidia.com/gpu: 1 # Required to use GPU.
+            ...
+
+Each of the Ray pods in the group can be scheduled on an AWS `p2.xlarge` instance (1 GPU, 4vCPU, 61Gi RAM).
+
+.. tip::
+
+    GPU instances are expensive -- consider setting up autoscaling for your GPU Ray workers,
+    as demonstrated with the `minReplicas:0` and `maxReplicas:5` settings above.
+    To enable autoscaling, remember also to set `enableInTreeAutoscaling:True` in your RayCluster's `spec`
+    Finally, make sure your group or pool of GPU Kubernetes nodes are configured to autoscale.
+
+GPUs and Ray
+____________
+
+The KubeRay operator advertises container GPU resource limits to
+the Ray scheduler and the Ray autoscaler. In particular, the Ray container's
+`ray start` entrypoint will be automatically configured with the appropriate `--num-gpus` option.
+
+* Learn more about Ray's :ref:`gpu-support`.
+
+GPU workload scheduling
+~~~~~~~~~~~~~~~~~~~~~~~
+After a Ray pod with access to GPU is deployed, it will
+be able to execute tasks and actors annotated with gpu requests.
+For example, the decorator `@ray.remote(num_gpus=1)` annotates a task or actor
+requiring 1 GPU.
+
+
+GPU autoscaling
+~~~~~~~~~~~~~~~
+The Ray autoscaler is aware of each Ray worker group's GPU capacity.
+Say we have a RayCluster configured as in the config snippet above:
+- There is a worker group of Ray pods with 1 unit of GPU capacity each
+- The Ray cluster does not currently have any workers from that group
+- `maxReplicas` for the group is at least 2
+
+Then the following Ray program will trigger upscaling of 2 GPU workers.
+
+.. code-block:: python
+
+    import ray
+
+    ray.init()
+
+    @ray.remote(num_gpus=1)
+    class GPUActor:
+        def say_hello(self):
+            print("I live in a pod with GPU access.")
+
+    # Request actor placement.
+    gpu_actors = [GPUActor.remote() for _ in range(2)]
+    # The following command will block until two Ray pods with GPU access are scaled
+    # up and the actors are placed.
+    ray.get([actor.say_hello.remote() for actor in gpu_actors])
+
+After the program exits, the actors will be garbage collected.
+The GPU worker pods will be scaled down after the idle timeout (60 seconds by default).
+If the GPU worker pods were running on an autoscaling pool of Kubernetes nodes, the Kubernetes
+nodes will be scaled down as well.
+
+Requesting GPUs
+~~~~~~~~~~~~~~~
+You can also make a :ref:`direct request to the autoscaler<ref-autoscaler-sdk-request-resources>` to scale up GPU resources.
+
+.. code-block:: python
+
+    import ray
+
+    ray.init()
+    ray.autoscaler.sdk.request_resources(bundles=[{"GPU": 1} * 2])
```

Comment:
`[{"GPU": 1} * 2]` doesn't work in Python3.7. Maybe you meant `[{"GPU": 1}] * 2`?

What's the plan for adding CI for these examples? I don't know how it works, but I do see the other doc sections use `literalinclude` to include a selection from a file inside the `source/<topic>/doc_code` directory.

Reply:
Thanks for pointing the correction!

About testing -- For now, I will do the test manually before merging.
We already have CI that covers the functionality.

I will however open an issue to track covering this exact example in the CI -- will have to come back to that later, since we're on a tight timeline at the moment.

Reply:
https://github.com/ray-project/ray/issues/27388

Reply:
Fixed, tracked, tested manually.